### PR TITLE
Orgs: move duplicate API calls to Redux

### PR DESCRIPTION
**Pull request by thisisobate**
**What this PR does / why we need it**:
This PR moves duplicate API calls to Redux

**Todos**

- [x] update `selectOrgPage` to use Redux instead
- [x] write tests for `setUserOrganization` thunk action creator

**Which issue(s) this PR fixes**:

Relates to #25117

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/org/NewOrgPage.tsx [hunk 1]
```diff
@@ -30,13 +24,21 @@ const validateOrg = async (orgName: string) => {
...
 
-export const NewOrgPage: FC<PropsWithState> = ({ navModel }) => {
+export const NewOrgPage: FC<PropsWithState> = ({ navModel, setUserOrganization }) => {
+  const createOrg = async (newOrg: { name: string }) => {
+    const result = await getBackendSrv().post('/api/orgs/', newOrg);
+
+    setUserOrganization(result.orgId);
+    window.location.href = getConfig().appSubUrl + '/org';
+  };
+
```
Comment (ashharrison90) [inline public/app/features/org/NewOrgPage.tsx:41]: do you think we can just move all the `createOrg` logic into the reducer? 🤔

Comment (thisisobate) [reply public/app/features/org/NewOrgPage.tsx:41]: yeah...sounds like a good idea to me

Comment (thisisobate) [reply public/app/features/org/NewOrgPage.tsx:41]: > move all the createOrg logic into the reducer

The reducer you mean here is all the logic in `org/state/action.ts` right? You know these terms can become a bit confusing sometimes 😅

**Review by natellium — APPROVED** [review]
LGTM

File: public/app/features/org/SelectOrgPage.tsx [hunk 2]
```diff
@@ -19,17 +21,23 @@ const navModel = {
...
-export const SelectOrgPage: FC = () => {
+const connector = connect(null, mapDispatchToProps);
+
+type Props = ConnectedProps<typeof connector>;
+
+export const SelectOrgPage: FC<Props> = ({ setUserOrganization }) => {
   const [orgs, setOrgs] = useState<UserOrg[]>();
 
+  const setUserOrg = async (org: UserOrg) => {
+    setUserOrganization(org.orgId);
```
Comment (Clarity-89) [inline public/app/features/org/SelectOrgPage.tsx:37]: This action does an async API request, but looking at the code, the page location will be changed possibly before the request is completed. One issue I noticed is, if for some reason changing org fails, the user will still be redirected to the root page, but for the wrong org. Also I think it's a bit inconsistent that we set location this time directly in component, but other time in redux action.

Comment (natellium) [reply public/app/features/org/SelectOrgPage.tsx:37]: Good call @Clarity-89 my morning brain totally skipped that! @thisisobate I think we should take another look at this.

Solve the async issue @Clarity-89 mentioned and rethink how we want to do the redirection. IMO I'd keep the action clean and do the redirection in the components.

Comment (thisisobate) [reply public/app/features/org/SelectOrgPage.tsx:37]: Good catch! @Clarity-89 would open a separate PR for this. @natellium I agree with you...makes sense for redirections to happen in the components

**Pull request merged**